### ISSUE TITLE
Best way to solve the "ControlPort Error"?

### ISSUE THREAD

**Issue by antonott** [bastibl/gr-ieee802-11#292]
I suppose this is more of a question than an issue. Please tell me if I should send/post this somewhere else instead.

## TLDR
What is a **simple** and **stable** way to get past the [ControlPort Error](https://github.com/gnuradio/gnuradio/issues/3673) to get `gr-ieee802-11` up and running with GNURadio 3.8 on Ubuntu 20.04.2 LTS? I don't mind uninstalling GNU Radio and reinstalling in a different way (from source? with PyBOMBS?) if that is a good way to go about it.

## Description
I am trying to set up `gr-ieee802-11`, `maint-3.8` branch, with GNU Radio 3.8, but when I try to run `wifi_loopback.grc` I get [this issue](https://github.com/gnuradio/gnuradio/issues/3673). I tried to solve it [this way](https://github.com/gnuradio/gnuradio/issues/3673#issuecomment-777135079), but it did not seem to make any difference. I suppose I need to try the instructions in [this comment](https://github.com/gnuradio/gnuradio/issues/3673#issuecomment-717519232) then, but I don't quite understand how that works.
- Am I supposed to build GNU Radio with the PR [#2739](https://github.com/gnuradio/gnuradio/pull/2739) applied? If so, what is a good way to do that?
- Should I also replace all occurences of `PlatformThreadFactory` with `ThreadFactory` in `thrift_server_template.h`?
- To do both of these things I need to build from source, right?
- Is there not a simpler/cleaner way?

## Details
I installed GNU Radio 3.8 with `apt` from `universe` repo under Ubuntu 20.04.2 LTS. Note: I did **not** use the PPA. Some additional details:
```bash
~$ python3 -V
Python 3.8.10
~$ thrift -version
Thrift version 0.13.0
~$ gnuradio-config-info -v
3.8.1.0
~$ gnuradio-config-info --prefix
/usr
```
The prefix is due to the distribution. As a result i have to use `cmake -DCMAKE_INSTALL_PREFIX=/usr ../` when building OOT modules, but I doubt that has any relevance here.

**Comment by antonott** [comment]
I have now done some more troubleshooting on my own by uninstalling everything and then reinstalling, but this time with PyBOMBS. To be precise i initialized the prefix with `gr-recipes/gnuradio-default.lwr` and then also installed `gr-recipes/gr-ieee-80211.lwr ` and `gr-recipes/apache-thrift.lwr`.

Now the wifi loopback flowgraph runs, but I get the following console output:
```bash
~/pybombs_prefixes/gnuradio-default$ source setup_env.sh
~/pybombs_prefixes/gnuradio-default$ gnuradio-companion
<<< Welcome to GNU Radio Companion v3.8.3.1-3-gada48d50 >>>

Block paths:
        /home/antonott/.grc_gnuradio
        /home/antonott/pybombs_prefixes/gnuradio-default/share/gnuradio/grc/blocks

Loading: "/home/antonott/pybombs_prefixes/gnuradio-default/src/gr-ieee-80211/examples/wifi_loopback.grc"
>>> Done

Generating: '/home/antonott/pybombs_prefixes/gnuradio-default/src/gr-ieee-80211/examples/wifi_loopback.py'
>>> Warning: This flow graph may not have flow control: no audio or RF hardware blocks found. Add a Misc->Throttle block to your flow graph to avoid CPU congestion.

Executing: /usr/bin/python3 -u /home/antonott/pybombs_prefixes/gnuradio-default/src/gr-ieee-80211/examples/wifi_loopback.py

MAPPER: encoding: 0
set_min_output_buffer on block 7 to 207744
set_min_output_buffer on block 9 to 2077440
set_min_output_buffer on block 11 to 2077440
set_min_output_buffer on block 12 to 207744
set_min_output_buffer on block 13 to 415488
set_min_output_buffer on block 14 to 207744
set_min_output_buffer on block 30 to 960000
ControlPort Monitor running.
MAPPER called offset: 0   length: 0
Failed to to connect to ControlPort. Please make sure that you have Thrift installed and check your firewall rules.
MAPPER called offset: 0   length: 0
MAPPER called offset: 0   length: 0
MAPPER: received new message
MAPPER called offset: 0   length: 672
MAPPER called offset: 0   length: 672
MAPPER called offset: 0   length: 672
MAPPER: received new message
MAPPER called offset: 0   length: 672
gr::log :INFO: parse_mac0 - length: 34

new mac frame  (length 34)
=========================================
duration: 00 00
frame control: 00 08 (DATA)
Subtype: Data
seq nr: 0
mac 1: 42:42:42:42:42:42
mac 2: 23:23:23:23:23:23
mac 3: ff:ff:ff:ff:ff:ff
instantaneous fer: 0
xxxxxxxxxx
MAPPER called offset: 0   length: 672
MAPPER called offset: 0   length: 672
MAPPER: received new message
MAPPER called offset: 0   length: 672
gr::log :INFO: parse_mac0 - length: 34
```
Identical (apart from the `seq nr`) `new mac frame` messages continue to be printed until I interrupt. This isn't what it is supposed to look like, right? I should see some decoded "Hello World" messages? The interesting part is
```bash
Failed to to connect to ControlPort. Please make sure that you have Thrift installed and check your firewall rules.
```
But I do have Thrift installed (in the gnuradio-default environment):
```bash
~/pybombs_prefixes/gnuradio-default$ which thrift
/home/antonott/pybombs_prefixes/gnuradio-default/bin/thrift
```
And the firewall thing seems like a red herring since this is all local, right?

I would be very grateful for any advice on how to get past this and get gr-ieee802-11 working with GNU Radio.

**Comment by antonott** [comment]
Oh, I see now that the messaging actually seems to be working as intended! I thought the "xxxxxxxxxx" messages printed in the console were due to an error with ControlPort/Thrift, but it turns out these were the literal messages being sent by the `Message Strobe`. So the combination of the readme saying I should expect "Hello World" and me seeing the message about ControlPort and Thrift made me assume there was something wrong with the messaging, without actually taking the time to inspect the flowgraph a little bit more carefully.

This still leaves me wondering what the `CtrlPort Performance Monitor` block is for and why it does not work, but maybe I can read up on that myself now that I am not completely lost.
